queue_services/activity-log-listener/src/activity_log_listener/worker.py

```diff
@@ -0,0 +1,76 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The unique worker functionality for this service is contained here.
+
+The entry-point is the **cb_subscription_handler**
+
+The design and flow leverage a few constraints that are placed upon it
+by NATS Streaming and using AWAIT on the default loop.
+- NATS streaming queues require one message to be processed at a time.
+- AWAIT on the default loop effectively runs synchronously
+
+If these constraints change, the use of Flask-SQLAlchemy would need to change.
+Flask-SQLAlchemy currently allows the base model to be changed, or reworking
+the model to a standalone SQLAlchemy usage with an async engine would need
+to be pursued.
+"""
+import json
+import os
+
+import nats
+from auth_api.models import ActivityLog as ActivityLogModel
+from auth_api.models import db
+from entity_queue_common.service import QueueServiceManager
+from entity_queue_common.service_utils import QueueException, logger
+from flask import Flask  # pylint: disable=wrong-import-order
+
+from activity_log_listener import config
+
+
+qsm = QueueServiceManager()  # pylint: disable=invalid-name
+APP_CONFIG = config.get_named_config(os.getenv('DEPLOYMENT_ENV', 'production'))
+FLASK_APP = Flask(__name__)
+FLASK_APP.config.from_object(APP_CONFIG)
+db.init_app(FLASK_APP)
+
+
+async def process_event(event_message, flask_app):
+    """Insert into Activity log table."""
+    if not flask_app:
+        raise QueueException('Flask App not available.')
+
+    with flask_app.app_context():
+        data = event_message.get('data')
```

Comment:
if the subject is same for both account events [currently used for freeze], shouldn't it need some filters here ?

Reply:
Nope..they are different now..I couldn't get the wildcard working